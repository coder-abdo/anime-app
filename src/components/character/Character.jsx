import React from "react";
import { CharacterContainer } from "./CharacterStyle";
export const Character = ({ image_url, role, name }) => {
  return (
    <CharacterContainer>
      <div className="imgContainer">
        <img src={image_url} alt={name} />
      </div>
      <footer>
        <h4 data-testid="character-name">name: {name}</h4>
        <h4 data-testid="character-role">role: {role}</h4>
      </footer>
    </CharacterContainer>
  );
};
