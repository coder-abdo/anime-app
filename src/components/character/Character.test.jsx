import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Character } from "./Character";
afterEach(cleanup);
test("render Charcter properley", () => {
  const mokeCharacter = {
    image_url: "just photo",
    name: "john name",
    role: "hero"
  };
  const { container, getByTestId, asFragment } = render(
    <Character {...mokeCharacter} />
  );
  expect(container).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
  expect(getByTestId("character-name").textContent).toBe("name: john name");
  expect(getByTestId("character-role").textContent).toBe("role: hero");
});
