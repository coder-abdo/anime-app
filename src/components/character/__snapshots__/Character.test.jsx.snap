// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Charcter properley 1`] = `
<DocumentFragment>
  <div
    class="sc-AxjAm bQfzZP"
  >
    <div
      class="imgContainer"
    >
      <img
        alt="john name"
        src="just photo"
      />
    </div>
    <footer>
      <h4
        data-testid="character-name"
      >
        name: john name
      </h4>
      <h4
        data-testid="character-role"
      >
        role: hero
      </h4>
    </footer>
  </div>
</DocumentFragment>
`;
